src/containers/Application/ApplicationSummary.tsx

```diff
@@ -13,24 +15,51 @@ interface ApplicationProps {
 }
 
 const ApplicationSummary: React.FC<ApplicationProps> = ({ structure }) => {
-  const { replace } = useRouter()
+  const { replace, push } = useRouter()
+  const [shouldRevalidate, setShouldRevalidate] = useState(false)
+  const [isRevalidated, setIsRevalidated] = useState(false)
+  const {
+    userState: { currentUser },
+  } = useUserState()
   const { error, fullStructure } = useGetFullApplicationStructure({
     structure,
+    shouldRevalidate,
+  })
+
+  const { error: submitError, processing, submitted, submit } = useSubmitApplication({
+    serialNumber: fullStructure?.info.serial as string,
+    currentUser: currentUser as User,
   })
 
   useEffect(() => {
-    if (!fullStructure) return
+    if (!fullStructure || submitted) return
     // Re-direct if application is not valid
     if (fullStructure.info.firstStrictInvalidPage) {
       const { sectionCode, pageNumber } = fullStructure.info.firstStrictInvalidPage
       replace(`/applicationNEW/${fullStructure.info.serial}/${sectionCode}/Page${pageNumber}`)
     }
+    if (shouldRevalidate) setIsRevalidated(true)
   }, [fullStructure])
 
-  const handleSubmit = () => {}
+  const handleSubmit = () => {
+    setShouldRevalidate(true)
+  }
+
+  useEffect(() => {
+    // This should only run after user has clicked Submit
+    // and revalidation has completed successfully
+    if (!fullStructure?.responsesByCode) return
+    submit(Object.values(fullStructure?.responsesByCode))
+  }, [isRevalidated])
+
+  useEffect(() => {
+    if (submitted) push(`/applicationNEW/${fullStructure?.info.serial}/submission`)
```

Comment:
if we wait for submission, what could happen is this:
* mutation fires up to change application trigger
* the first getLoadApllicationNEW re-fires because getApplicationNEW query refetched (hopefully)
* then loading is set to true
* the component tree is now gone this component doesn't exist anymore
* when loading is finished components are rebuild and this component has new state (so newer really => submits).

I think you should call navigation to submission straight after the submit or await for submit and call navigation (can await apolo mutations btw, not sure that useSubmitApplication.submit is though)